local_planner/src/nodes/histogram.cpp

```diff
@@ -52,17 +53,10 @@ void Histogram::downsample() {
   dist_ = temp_dist;
 }
 
-void Histogram::setZero() { dist_.fill(0.f); }
-
-bool Histogram::isEmpty() const {
-  int counter = 0;
-  for (int e = 0; e < e_dim_; e++) {
-    for (int z = 0; z < z_dim_; z++) {
-      if (dist_(e, z) > FLT_MIN) {
-        counter++;
-      }
-    }
-  }
-  return counter == 0;
+void Histogram::setZero() {
+  dist_.fill(0.f);
+  is_empty_ = true;
 }
+
+bool Histogram::isEmpty() const { return is_empty_; }
```

Comment:
Correct me if I am wrong, but I think this will never become false with the current implementation.

Reply:
yes that was exactly was I was trying to say :+1:

Reply:
Since the cell may set to 0.f. So it needs check all cells in Histogram::isEmpty(), rollback to early stopping version.